dask/dataframe/partitionquantiles.py

```diff
@@ -427,7 +427,7 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
     try:
         vals = data.quantile(q=qs / 100, interpolation=interpolation).values
     except (TypeError, NotImplementedError):
-        vals, _ = _percentile(array_safe(data, data.dtype), qs, interpolation)
+        vals, _ = _percentile(array_safe(data, like=data.values), qs, interpolation)
```

Comment:
I think you're correct that cudf would fail either way. However, is this fall-back currently covered by any pandas-based tests?

Reply:
Yeah, there are a couple shuffle tests (such as [`test_set_index_general`](https://github.com/dask/dask/blob/b31c7cf60eb06aad52f3691d2a7ab784fd7cc039/dask/dataframe/tests/test_shuffle.py#L190-L215)) that shuffle over a string column and cover this fallback, I think this just didn't raise any alarms because [`_array_like_safe`](https://github.com/dask/dask/blob/b31c7cf60eb06aad52f3691d2a7ab784fd7cc039/dask/array/utils.py#L425-L438) and `np.array` don't raise any errors/warnings if `like` isn't array-like.